test/clientconn_test.go

```diff
@@ -180,39 +162,56 @@ func (s) TestClientConnRPC_WithoutNameResolutionDelay(t *testing.T) {
 // nameResolutionDelayed flag is set indicating there was a delay in name
 // resolution waiting for resolver to return addresses.
 func (s) TestClientConnRPC_WithNameResolutionDelay(t *testing.T) {
-	stub, cleanup := startStubServer(t)
-	defer cleanup()
+	stub := &stubserver.StubServer{
+		EmptyCallF: func(ctx context.Context, in *testpb.Empty) (*testpb.Empty, error) {
+			return &testpb.Empty{}, nil
+		},
+	}
+	if err := stub.Start(nil); err != nil {
+		t.Fatalf("Failed to start StubServer: %v", err)
+	}
+	defer stub.Stop()
 
 	statsHandler := &testStatsHandler{}
-	cc, rb := createTestClient(t, "delayed", statsHandler)
+	rb := manual.NewBuilderWithScheme("delayed")
+	cc, err := grpc.NewClient(rb.Scheme()+":///test.server",
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+		grpc.WithResolvers(rb),
+		grpc.WithStatsHandler(statsHandler),
+	)
+	if err != nil {
+		t.Fatalf("grpc.NewClient() failed: %v", err)
+	}
 	defer cc.Close()
 
-	client := testgrpc.NewTestServiceClient(cc)
-	rpcError := make(chan error, 1)
+	time.AfterFunc(defaultTestShortTimeout, func() {
+		rb.UpdateState(resolver.State{Addresses: []resolver.Address{{Addr: stub.Address}}})
+	})
+
 	ctx, cancel := context.WithTimeout(context.Background(), defaultTestTimeout)
 	defer cancel()
 
-	go func() {
-		t.Log("RPC waiting for resolved addresses")
-		_, err := client.EmptyCall(ctx, &testpb.Empty{})
-		rpcError <- err
-	}()
+	client := testgrpc.NewTestServiceClient(cc)
+	if _, err := client.EmptyCall(ctx, &testpb.Empty{}); err != nil {
+		t.Fatalf("EmptyCall RPC failed: %v", err)
+	}
 
-	timer := time.AfterFunc(100*time.Millisecond, func() {
-		rb.UpdateState(resolver.State{Addresses: []resolver.Address{{Addr: stub.Address}}})
-		t.Log("Resolver updated to return addresses.")
-	})
-	defer timer.Stop()
+	if !statsHandler.nameResolutionDelayed {
+		t.Fatalf("Expected nameResolutionDelayed to be true, got false")
```

Comment:
nit: got before want please

Reply:
If you're doing that, then it should be more like:

```suggestion
		t.Fatalf("statsHandler.nameResolutionDelayed = false; want true")
```

Reply:
Done